devito/functions/basic.py

```diff
@@ -66,7 +66,7 @@ class Basic(object):
     is_Dimension = False
     is_Constant = False
     # Tensor symbolic objects created by the user
-    is_TensorFunction = False
+    is_GridedFunction = False
```

Comment:
GridFunction, rather than GridedFunction?

Reply:
`DiscreteFunction`?

Reply:
the all necessarily are Discrete...

The reason I like Grid/Grided is that a TensorFunction, unlike it's superclasses, is aware of the concept of Grid. Now, Grid sounds better than Grided.

Said that, I do not really dislike Discrete. So if you don't like GridFunction, feel free to use DiscreteFunction instead.